### DISCUSSION TITLE
How is the compiler supposed to be installed?

### DISCUSSION THREAD

**Discussion by paramsiddharth** [google/closure-compiler#4003] [Q&A] [answered]
Hello everyone! I discovered the Closure Library last year, and have been looking to try out the Closure Compiler out locally.

From what I understood, I can download the JAR from Maven and use it with Java in the command-line, but the README in this repository tells otherwise – It says:

![image](https://user-images.githubusercontent.com/30315706/198049324-b9aa514a-2e83-4112-b9eb-7cbcf3f759b0.png)

However, the [NPM page](https://www.npmjs.com/package/google-closure-compiler) says:

![image](https://user-images.githubusercontent.com/30315706/198049559-322b5093-362d-4882-8f31-5ab15152942f.png)

I am already aware of the fact that the JavaScript version of the Closure Compiler is no longer maintained, but in my past I've come across libraries on NPM that can be installed globally and are not actually made in JavaScript. Is the package currently published on NPM now a wrapper for the Java version? If not, should we edit the README to explicitly say that the NPM version is deprecated?

Thank you!

**Comment by brad4d** [comment, marked as answer]
When you install via NPM you should generally end up getting a version of the compiler that has been run through the Graal tool to turn it into a native binary.

The NPM distribution of closure-compiler is handled by https://github.com/google/closure-compiler-npm
It is not deprecated.

You may want to create an issue over there if you have further questions.

**Reply by paramsiddharth** [reply]
Thank you, Bradford. The NPM page listing says it is deprecated, which is why I had the question.

**Comment by brad4d** [comment]
http://go/closure-compiler-npm-pr/268 will hopefully update the confusing message.
